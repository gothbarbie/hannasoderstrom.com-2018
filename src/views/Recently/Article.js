// @flow
import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'

import styled from 'styled-components'

import { fetchRecently } from '../../actions'

import Wrapper from '../../components/Layout/Wrapper'
import Small from '../../components/Small'
import LoadingSpinner from '../../components/LoadingSpinner'

const WrapperSmall = styled.section`
  max-width: 600px;
  margin: 5% auto;
  padding: 0 2rem;
`

const H1 = styled.h1`
  font-size: 2rem;
  font-weight: bold;
`

type Props = {
  loading: boolean,
  fetchRecently: Function,
  recentlies: Array<{}>,
}

class Article extends Component<Props> {
  componentDidMount () {
    const url = decodeURI(window.location.pathname).split('/')
    const slug = url[2]
    const regex = new RegExp(/([A-Za-z0-9\-])+/g)

    if (regex.test(decodeURI(slug))) {
      this.props.fetchRecently(slug)
    }
  }

  render () {
    if (this.props.loading) return <LoadingSpinner />

    if (!this.props.recentlies.length) return <p>Not found.</p>

    const { content, title } = this.props.recentlies[0]

    return (
      <Wrapper>
        <Link to="/recently">‚Üê Back</Link>
          <WrapperSmall>
            <H1 dangerouslySetInnerHTML={{ __html: title.rendered }} />
              <div dangerouslySetInnerHTML={{ __html: content.rendered }} />
          </WrapperSmall>
      </Wrapper>
    )
  }
}

const mapStateToProps = ({ recentlies, loading }) => {
  return {
    recentlies,
    loading,
  }
}

const mapDispatchToProps = {
  fetchRecently,
}

export default connect(mapStateToProps, mapDispatchToProps)(Article)
