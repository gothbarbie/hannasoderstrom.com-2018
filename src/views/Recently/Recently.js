import React from 'react'
import styled from 'styled-components'

import Wrapper from '../../components/Layout/Wrapper'
import Paragraph from '../../components/Paragraph'

const Grid = styled.section`
  width: 100%;
  display: grid;
  padding: 2rem 0;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-rows: 2fr;
  grid-column-gap: 5%;
  grid-row-gap: 10%;
  align-items: start;

  & img {
    min-height: 100px;
  }

  .first {
    grid-column: 1 / span 2;
    grid-row: 1 / span 2;

    & img {
      min-height: 200px;
    }
  }
`

const Article = styled.article`
  margin: 1rem 0;
`

const H2 = styled.h2`
  font-size: 1.5rem;
  font-weight: 500;
  margin: 1rem 0;
`

const Img = styled.img`
  width: 100%;
  height: auto;
  background: #efefef;
`

const Recently = () => (
  <Wrapper>
    <Grid>
      <Article className="first">
        <Img />
        <H2>New portfolio design</H2>
        <Paragraph>
          I try to re-do my portfolio from time to time. It’s a good project for
          trying out new technologies, if nothing else. It also means that I get
          to improve my design skills, since I don’t do much designing now a
          days (focusing mostly on...
        </Paragraph>
      </Article>
      <Article>
        <Img />
        <H2>New portfolio design</H2>
        <p>Introduction text...</p>
      </Article>
      <Article>
        <Img />
        <H2>New portfolio design</H2>
        <p>Introduction text...</p>
      </Article>
      <Article>
        <Img />
        <H2>New portfolio design</H2>
        <p>Introduction text...</p>
      </Article>
      <Article>
        <Img />
        <H2>New portfolio design</H2>
        <p>Introduction text...</p>
      </Article>
      <Article>
        <Img />
        <H2>New portfolio design</H2>
        <p>Introduction text...</p>
      </Article>
      <Article>
        <Img />
        <H2>New portfolio design</H2>
        <p>Introduction text...</p>
      </Article>
    </Grid>
  </Wrapper>
)

export default Recently
