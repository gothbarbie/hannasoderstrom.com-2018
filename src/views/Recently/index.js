import Recently from './Recently'
export default Recently
