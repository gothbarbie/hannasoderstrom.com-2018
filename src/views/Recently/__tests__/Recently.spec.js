import React from 'react'
import { shallow } from 'enzyme'

import { Recently } from '../Recently'

describe('views/Recently', () => {
  let component
  let store
  let fetchRecentlies

  beforeEach(() => {
    fetchRecentlies = jest.fn()
    store = {
      dispatch: jest.fn(),
      getState: jest.fn(),
      subscribe: jest.fn(),
    }
    component = shallow(
      <Recently
        fetchRecentlies={fetchRecentlies}
        recentlies={[]}
        store={store}
      />
    )
  })

  it('renders when no content', () => {
    expect(component).toMatchSnapshot()
  })

  it('renders with content', () => {
    component.setProps({
      recentlies: [
        {
          id: 1,
          title: { rendered: 'Hello' },
          excerpt: {
            rendered: '<p>Hello</p>',
          },
          date: 'today',
          content: { rendered: '<p>Hello world</p>' },
          media: {
            guid: {
              rendered: 'http://www.image.com/image01.jpg',
            },
          },
        },
      ],
    })
    expect(component).toMatchSnapshot()
  })

  describe('#fetchRecentlies', () => {
    it('has been called on mount', () => {
      expect(fetchRecentlies).toHaveBeenCalled()
    })
  })
})
