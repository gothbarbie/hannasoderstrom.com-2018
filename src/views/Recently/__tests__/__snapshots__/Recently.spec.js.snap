// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/Recently renders when no content 1`] = `
<styled.div>
  <styled.section />
</styled.div>
`;

exports[`views/Recently renders with content 1`] = `
<styled.div>
  <styled.section>
    <styled.article
      className="first"
      data-test="article"
      key="1"
    >
      <Styled(Link)
        to="recently/undefined"
      >
        <styled.h2
          dangerouslySetInnerHTML={
            Object {
              "__html": "Hello",
            }
          }
        />
        <img
          alt="Hello"
          src="http://www.image.com/image01.jpg"
        />
        <styled.small
          dangerouslySetInnerHTML={
            Object {
              "__html": "Invalid Date",
            }
          }
        />
        <main
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Hello</p>",
            }
          }
        />
      </Styled(Link)>
    </styled.article>
  </styled.section>
</styled.div>
`;
